diff --git i/index.js w/index.js
index 19a6031e..b6a247bb 100644
--- i/index.js
+++ w/index.js
@@ -35,7 +35,7 @@ function loadScript(scriptUrl) {
 const storeKey = "store";

 // Remove trailing slash from root because it's used by the Elm API to resolve backend api urls
-const clientUrl = location.origin + (location.pathname == "/" ? "" : location.pathname);
+const clientUrl = (location.origin + location.pathname).replace(/\/+$/g, "");

 const app = Elm.Main.init({
   flags: {
