diff --git i/src/Data/Github.elm w/src/Data/Github.elm
index 9d73fb78..4cf26581 100644
--- i/src/Data/Github.elm
+++ w/src/Data/Github.elm
@@ -27,6 +27,7 @@ type alias Release =
     , hash : String
     , markdown : String
     , name : String
+    , prerelease : Bool
     , tag : String
     , url : String
     }
@@ -60,6 +61,7 @@ decodeRelease =
         |> Pipe.required "target_commitish" Decode.string
         |> Pipe.required "body" Decode.string
         |> Pipe.required "name" Decode.string
+        |> Pipe.required "prerelease" Decode.bool
         |> Pipe.required "tag_name" Decode.string
         |> Pipe.required "html_url" Decode.string

@@ -67,10 +69,17 @@ decodeRelease =
 decodeReleaseList : Decoder (List Release)
 decodeReleaseList =
     Decode.list decodeRelease
-        -- Exclude draft releases
-        |> Decode.andThen (List.filter (.draft >> not) >> Decode.succeed)
+        -- Exclude draft and pre-releases
+        |> Decode.map (List.filter (\{ draft, prerelease } -> not draft && not prerelease))


 unreleased : Release
 unreleased =
-    Release True "" "" "Unreleased" "Unreleased" ""
+    { draft = True
+    , hash = ""
+    , markdown = ""
+    , name = "Unreleased"
+    , prerelease = True
+    , tag = "Unreleased"
+    , url = ""
+    }
